zhaquirks/tuya/ts000f_switch.py

```diff
@@ -0,0 +1,91 @@
+"""tuya TS000F xkap8wtb On/Off In-line Switches with metering support."""
+
+from zigpy.profiles import zha
+from zigpy.zcl.clusters.general import (
+    Basic,
+    GreenPowerProxy,
+    Groups,
+    Identify,
+    OnOff,
+    Ota,
+    Scenes,
+    Time,
+)
+from zigpy.zcl.clusters.lightlink import LightLink
+
+from zhaquirks.const import (
+    DEVICE_TYPE,
+    ENDPOINTS,
+    INPUT_CLUSTERS,
+    MODEL,
+    OUTPUT_CLUSTERS,
+    PROFILE_ID,
+)
+from zhaquirks.tuya import (
+    TuyaZBE000Cluster,
+    TuyaZBElectricalMeasurement,
+    TuyaZBMeteringCluster,
+    TuyaZBOnOffAttributeCluster,
+)
+from zhaquirks.tuya.mcu import EnchantedDevice
+
+
+class Tuya_1G_Wall_Switch_Metering(EnchantedDevice):
+    """Tuya 1 gang wall switch with metering and restore power state support."""
+
+    signature = {
+        MODEL: "TS000F",
+        ENDPOINTS: {
+            # SizePrefixedSimpleDescriptor(endpoint=242, profile=41440, device_type=97, device_version=0, input_clusters=[], output_clusters=[33])
+            # input_clusters=["0x0000","0x0003","0x0004","0x0005","0x0006", "0x000a", "0x0702","0x0b04","0x1000","0x1888","0xe000"]
+            # output_clusters=["0x0019"]>
+            1: {
+                PROFILE_ID: zha.PROFILE_ID,
+                DEVICE_TYPE: zha.DeviceType.ON_OFF_LIGHT,
+                INPUT_CLUSTERS: [
+                    Basic.cluster_id,
+                    Identify.cluster_id,
+                    Groups.cluster_id,
+                    Scenes.cluster_id,
+                    OnOff.cluster_id,
+                    Time.cluster_id,
+                    TuyaZBMeteringCluster.cluster_id,
+                    TuyaZBElectricalMeasurement.cluster_id,
+                    LightLink.cluster_id,
+                    0x1888,
+                    TuyaZBE000Cluster.cluster_id,
+                ],
+                OUTPUT_CLUSTERS: [Ota.cluster_id],
+            },
+            # <SimpleDescriptor endpoint=242 profile=41440 device_type=97
+            # input_clusters=[]
+            # output_clusters=["0x0021"]>
+            242: {
+                PROFILE_ID: 41440,
+                DEVICE_TYPE: 97,
+                INPUT_CLUSTERS: [],
+                OUTPUT_CLUSTERS: [GreenPowerProxy.cluster_id],
+            },
+        },
+    }
+    replacement = {
+        ENDPOINTS: {
+            1: {
+                PROFILE_ID: zha.PROFILE_ID,
+                DEVICE_TYPE: zha.DeviceType.ON_OFF_PLUG_IN_UNIT,
```

Comment:
If the device is working OK with the original device type it shall not getting changed then some device dont like it and getting problems in the long run.
Change the device type on the device card or in device config.yaml so the Zigbee type is not making problems.

Reply:
Actually as stated in [issue/2447](https://github.com/zigpy/zha-device-handlers/issues/2447) and observed in field the original device type detected is wrong (light), so we set it right here?..

Reply:
Its not nice done from tuya but is the right way do in ZHA for not getting one system that have problems so in the replasment is shall being `DEVICE_TYPE: zha.DeviceType.ON_OFF_LIGHT,`.
And use the method Erkr have linked in the issue https://github.com/zigpy/zha-device-handlers/issues/2447#issuecomment-1742132942.

Reply:
@MattWestb patched in [4009eca](https://github.com/zigpy/zha-device-handlers/pull/2613/commits/4009eca17f801f5de6e0228dc15d31b4823bc854).